docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,474 @@
+```
+
+Transform the `Test` data by adding the following code to `EvaluateModel()`:
+[!code-csharp[Transform](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#Transform "Transform the test data")]
+
+The [Transform()](xref:Microsoft.ML.ITransformer.Transform%2A) method makes predictions for multiple provided input rows of a test dataset.
+
+Evaluate the model by adding the following as the next line of code in the `EvaluateModel()` method:
+
+[!code-csharp[Evaluate](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#Evaluate "Evaluate the model using predictions from the test data")]
+
+Once you have the prediction set, the [Evaluate()](xref:Microsoft.ML.RecommendationCatalog.Evaluate%2A) method assesses the model, which compares the predicted values with the actual `Labels` in the test dataset and returns metrics on how the model is performing.
+
+Print your evaluation metrics to the console by adding the following as the next line of code in the `EvaluateModel()` method:
+
+[!code-csharp[PrintMetrics](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#PrintMetrics "Print the evaluation metrics")]
+
+This output should look similar to the following:
+
+```console
+=============== Training the model ===============
+iter      tr_rmse          obj
+   0       1.5403   3.1262e+05
+   1       0.9221   1.6030e+05
+   2       0.8687   1.5046e+05
+   3       0.8416   1.4584e+05
+   4       0.8142   1.4209e+05
+   5       0.7849   1.3907e+05
+   6       0.7544   1.3594e+05
+   7       0.7266   1.3361e+05
+   8       0.6987   1.3110e+05
+   9       0.6751   1.2948e+05
+  10       0.6530   1.2766e+05
+  11       0.6350   1.2644e+05
+  12       0.6197   1.2541e+05
+  13       0.6067   1.2470e+05
+  14       0.5953   1.2382e+05
+  15       0.5871   1.2342e+05
+  16       0.5781   1.2279e+05
+  17       0.5713   1.2240e+05
+  18       0.5660   1.2230e+05
+  19       0.5592   1.2179e+05
+=============== Evaluating the model ===============
+Rms: 0.994051469730769
+RSquared: 0.412556298844873
+```
+
+In this output, there are 20 iterations. In each iteration, the measure of error decreases and converges closer and closer to 0.
+
+The `root of mean squared error` (RMS or RMSE) is frequently used to measure the differences between values predicted by a model and the values actually observed in a test dataset. Technically it's the square root of the average of the squares of the errors. You want your RMSE score to be as close to 1 as possible.
+
+`R Squared` is the variation percentage in the predicted values explained by your model. It's a value between 0 and 1, and the closer the value is to 0, the better the model is.
+
+Add the following as the next line of code in the `Main()` method to call your `EvaluateModel()` method:
+
+[!code-csharp[EvaluateModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#EvaluateModelMain "Add EvaluateModel method in Main")]
```

Comment:
Do we need to move this up above the output sections? i.e. we won't see the output until this method is called.